frontend/packages/console-shared/src/utils/firehose.ts

```diff
@@ -0,0 +1,39 @@
+import { FirehoseResult } from '@console/internal/components/utils';
+import { K8sKind, K8sResourceCommon, K8sResourceKind } from '@console/internal/module/k8s';
+import * as _ from 'lodash';
+
+export const isLoaded = <A extends K8sResourceCommon | K8sResourceCommon[] = K8sResourceKind[]>(
+  result: FirehoseResult<A>,
+) => result && result.loaded;
+
+export const getLoadedData = <
+  A extends K8sResourceCommon | K8sResourceCommon[] = K8sResourceKind[]
+>(
+  result: FirehoseResult<A>,
+  defaultValue: A = null,
+): A => (result && result.loaded && !result.loadError ? result.data : defaultValue);
+
+export const getLoadError = <A extends K8sResourceCommon | K8sResourceCommon[] = K8sResourceKind[]>(
```

Comment:
we should consider putting this directly into firehose so you would get nicer error message when accessing `loadError` prop

Reply:
@rawagner When adding this to `Firehose` - do you mean adding a new prop (to be passed to the given component) which represents a humanized error msg?

Reply:
could be done partially but not really possible, because this takes care also about the case when the FirehoseResult is null

keeping in kubevirt for now